JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/timeseries/TimeSeriesSparql.java

```diff
@@ -41,27 +42,107 @@
+	public static final String NS_TIME = "http://www.w3.org/2006/time#";
+
 	// Prefixes
-	private static final Prefix prefix_ontology = SparqlBuilder.prefix("ts", iri(ns_ontology));
-	private static final Prefix prefix_kb = SparqlBuilder.prefix("kb", iri(ns_kb));
-	
-	// RDF type
-    private static final Iri TimeSeries = prefix_ontology.iri("TimeSeries");
-    
-    // Relationships
-    private static final Iri hasTimeSeries = prefix_ontology.iri("hasTimeSeries");
-    private static final Iri hasRDB = prefix_ontology.iri("hasRDB");
-    private static final Iri hasTimeUnit = prefix_ontology.iri("hasTimeUnit");
-
-    // Fields for class specific exceptions
+	private static final Prefix PREFIX_ONTOLOGY = SparqlBuilder.prefix("ts", iri(TIMESERIES_NAMESPACE));
+	private static final Prefix PREFIX_KB = SparqlBuilder.prefix("kb", iri(TIMESERIES_NAMESPACE));
+	private static final Prefix PREFIX_TIME = SparqlBuilder.prefix("time", iri(NS_TIME));
+
+	// Relationships
+	private static final Iri hasTimeSeries = PREFIX_ONTOLOGY.iri("hasTimeSeries");
+	private static final Iri hasRDB = PREFIX_ONTOLOGY.iri("hasRDB");
+	private static final Iri hasTimeUnit = PREFIX_ONTOLOGY.iri("hasTimeUnit");
+	private static final Iri hasAveragingPeriod = PREFIX_ONTOLOGY.iri("hasAveragingPeriod");
+	private static final Iri numericDuration = PREFIX_TIME.iri("numericDuration");
+	private static final Iri unitType = PREFIX_TIME.iri("unitType");
+
+	// Fields for class specific exceptions
 	private final String exceptionPrefix = this.getClass().getSimpleName() + ": ";
 
-    /**
+	//Time Series Type
+	public static final String AVERAGE_TYPE_STRING = TIMESERIES_NAMESPACE+"AverageTimeSeries";
+	public static final String STEPWISE_CUMULATIVE_TYPE_STRING = TIMESERIES_NAMESPACE+"StepwiseCumulativeTimeSeries";
+	public static final String CUMULATIVE_TOTAL_TYPE_STRING = TIMESERIES_NAMESPACE+"CumulativeTotalTimeSeries";
+	public static final String INSTANTANEOUS_TYPE_STRING = TIMESERIES_NAMESPACE+"InstantaneousTimeSeries";
+	public static final String TIMESERIES_TYPE_STRING = TIMESERIES_NAMESPACE+"TimeSeries";
+
+	// RDF type
+	public static final Iri TIMESERIES = iri(TIMESERIES_TYPE_STRING);
+	public static final Iri STEPWISE_CUMULATIVE_TIMESERIES = iri(STEPWISE_CUMULATIVE_TYPE_STRING);
+	public static final Iri CUMULATIVE_TOTAL_TIMESERIES = iri(CUMULATIVE_TOTAL_TYPE_STRING);
+	public static final Iri AVERAGE_TIMESERIES = iri(AVERAGE_TYPE_STRING);
+	public static final Iri INSTANTANEOUS_TIMESERIES = iri(INSTANTANEOUS_TYPE_STRING);
+	private static final Iri Duration = PREFIX_TIME.iri("Duration");
+
+	//Values Pattern for TimeSeries types
+	private static final List<RdfObject> types = Arrays.asList(TIMESERIES, CUMULATIVE_TOTAL_TIMESERIES, STEPWISE_CUMULATIVE_TIMESERIES, AVERAGE_TIMESERIES, INSTANTANEOUS_TIMESERIES);
+	private static final Variable tsType = Queries.SELECT().var();
+	private static final ValuesPattern vp = new ValuesPattern(tsType, types);
```

Comment:
This is slightly risky because of how the query builder works, it starts from ?x0, ?x1, and so on for each SelectQuery object. If some method has another SelectQuery object and uses this, it will have 2 ?x0. Better put this within a method